### ISSUE TITLE
DISTINCT() AS with 2+ columns gives an "anonymous composite type" error message

### ISSUE THREAD

**Issue by colton-citus** [citusdata/citus#2434]
Following error occurs when attempting to SELECT DISTINCT() AS on a distributed table with more than one column within the DISTINCT clause: 

ERROR:  input of anonymous composite types is not implemented 

Minimum reproduction SQL script: 

```
create table t1 (col1 int,col2 int);
select create_distributed_table('t1','col1');
insert into t1(col1,col2) VALUES (1,1),(2,2),(1,2);
select distinct (col1,col2) AS foo FROM t1;
drop table t1;
```

Please note that a regular DISTINCT clause works fine here.

**Comment by marcocitus** [comment]
Somehow this query deparses to `SELECT DISTINCT ROW(col1, col2) AS foo FROM t1_102393 t1 WHERE true`, which seems like a bug in ruleutils.

**Comment by marcocitus** [comment]
this also applies to router queries

**Comment by marcocitus** [comment]
It's probably not a bug in ruleutils, but rather the inability of Postgres to parse anonymous composite types when the coordinator reads the results from the workers.

**Comment by marcocitus** [comment]
Minimal test case:
```
SELECT (col1,col2) FROM t1 WHERE col1 = 1;
ERROR:  input of anonymous composite types is not implemented
```

**Comment by marcocitus** [comment]
In Postgres itself, this limitation also manifests itself when the type of the record needs to be resolved.
```
CREATE TABLE foo AS SELECT (x,y) FROM local;
ERROR:  column "row" has pseudo-type record
```

**Comment by onderkalaci** [comment]
Looks like related with #2184 as well.

A nasty workaround to have the output have a type and skip the error: `select distinct ARRAY[col1,col2] AS foo FROM t1;`

**Comment by serprex** [comment]
Fixed by #3191
